htlcswitch/link.go

```diff
@@ -1601,9 +1639,10 @@ func (l *channelLink) handleDownstreamUpdateAdd(pkt *htlcPacket) error {
 		return errors.New("not an UpdateAddHTLC packet")
 	}
 
-	// If we are flushing the link in the outgoing direction we can't add
-	// new htlcs to the link and we need to bounce it
-	if l.IsFlushing(Outgoing) {
+	// If we are flushing the link in the outgoing direction or we have
+	// already sent Stfu, then we can't add new htlcs to the link and we
+	// need to bounce it.
+	if l.IsFlushing(Outgoing) || !l.quiescer.canSendUpdates() {
```

Comment:
Is `FailPermanentChannelFailure` the right error to be throwing here? I'd think that we want a `TemporaryChannelFailure` so that nodes sending through us don't give up on the channel entirely?